### PR TITLE
Day6

### PR DESCRIPTION
**Pull request by kghite**

### CODE REVIEW

**Comment by thecardkid** [comment]
Looks good. Although I think to agree on more scaffolding in the DS part, so could you either:
- embellish the readme to include what each function should accomplish, or
- add back in some of the methods you took out?

And then update the PR?

**Comment by davidabrahams** [comment]
I assume the methods Katie took out are the ones Allen intended them to
fill in?

**Comment by davidabrahams** [comment]
I fixed the README

File: day6/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Chapter 17
+
```
Comment (ssingal05) [inline day6/README.md]: Missing in "equals" in the README

File: day6/src/MyTreeMap.java [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+		// some implementations can handle null as a key, but not this one
+		if (target == null) {
+            throw new NullPointerException();
+	    }
+		
+		// something to make the compiler happy
+		@SuppressWarnings("unchecked")
+		Comparable<? super K> k = (Comparable<? super K>) target;
+		
+		// TODO: Your solution here
```
Comment (ssingal05) [inline day6/src/MyTreeMap.java]: Be consistent about
"Your solution here"
vs
"Your code here"

**Pull request merged**